plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/utils/ResourceProcessor.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.utils;
+
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistry;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.core.connection.ConnectionProperties;
+
+import com.openshift.restclient.OpenShiftException;
+import com.openshift.restclient.model.IResource;
+
+/**
+ * A Resource processor allows to plug resource specific actions.
+ * 
+ * @author Jeff Maury
+ *
+ */
+public interface ResourceProcessor {
+    /**
+     * Return if delete cascade will delete linked resources.
+     * 
+     * @return false if cascade delete is a simple delete
+     */
+    default boolean willCascadeDeleteLinkedResources() {
+        return false;
+    }
+    
+    /**
+     * Process the deletion of the resource.
+     * 
+     * @param connection the Openshift connection
+     * @param resource the resource to delete
+     * @param willDeleteSubResources if cascade delete
+     */
+    default void handleDelete(ConnectionsRegistry registry, Connection connection, IResource resource, boolean willDeleteSubResources) throws OpenShiftException {
```

Comment:
Since this method may be long running while deleting multiple resources, I suggest adding parameter IProgressMonitor and reporting to it large pieces of job, like requesting resources, deleting resources, waits for result returned by the client, etc.

By the way, may user have the rigth to cancel? If it is highly undesirable, a warning may pop up explaining all disadvantages of cancelling, but with the option to cancel anyway available. A cancel may happen in different forms anyway, like Eclipse crash, network disconnection, electric power shut-down, coffee spilled, etc. If by some reason action takes so long that user may decide that it is better to have one of the mentioned above than wait, it is better to give him the option to cancel.